src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,342 @@
+        [SOL_NETCTL_SERVICE_STATE_ONLINE] = "Online",
+        [SOL_NETCTL_SERVICE_STATE_DISCONNECT] = "Disconnect",
+        [SOL_NETCTL_SERVICE_STATE_FAILURE] = "Failure",
+        [SOL_NETCTL_SERVICE_STATE_REMOVE] = "Remove",
+    };
+
+    service_list = sol_netctl_get_services();
+
+    while (service = sol_ptr_vector_get(service_list,i)) {
+        state = sol_netctl_service_get_state(service);
+        name = sol_netctl_service_get_name(service);
+
+        if (name && strcmp(name, mdata->service_name) == 0) {
+            if (state == SOL_NETCTL_SERVICE_STATE_IDLE && connect) {
+                sol_netctl_service_connect(service);
+            } else if ((state == SOL_NETCTL_SERVICE_STATE_READY) && (!connect)) {
+                  sol_netctl_service_disconnect(service);
+            }
+            state = sol_netctl_service_get_state(service);
+            r = sol_flow_send_string_packet(node,
+                    SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__OUT,
+                    state_msgs[state]);
+            if (r < 0)
+                SOL_WRN("Failed to send network service connect state");
+            return 0;
+        }
+        i++;
+    }
+
+    sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+
+    return 0;
+}
+
+static int
+connect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return connection_handle(node, data, true);
+}
+
+static int
+disconnect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return connection_handle(node, data, false);
+}
+
+static int
+open_service_connection(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct connection_service_data *mdata = data;
+    const struct sol_flow_node_type_netctl_service_options *opts;
+    static const char *errmsg = "Could not add service monitor";
+
+    SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options,
+        SOL_FLOW_NODE_TYPE_NETCTL_SERVICE_OPTIONS_API_VERSION,
+        -EINVAL);
+    opts = (const struct sol_flow_node_type_netctl_service_options *)
+        options;
+
+    if (opts->name) {
+        mdata->service_name = strdup(opts->name);
+        SOL_NULL_CHECK(mdata->service_name, -ENOMEM);
+    }
+
+    if (sol_netctl_add_service_monitor(NULL, NULL))
+        goto error;
+
+    sol_netctl_add_error_monitor(error_cb, mdata);
+
+    return 0;
+
+error:
+    sol_flow_send_error_packet(node, EINVAL, "%s", errmsg);
+    SOL_WRN("%s", errmsg);
+
+    return -EINVAL;
+
+}
+
+static void
+close_service_connection(struct sol_flow_node *node, void *data)
+{
+    static const char *errmsg = "Could not delete service monitor";
+    struct connection_service_data *mdata = data;
+
+    free(mdata->service_name);
+
+    if (sol_netctl_del_service_monitor(NULL, NULL))
+        goto error;
```

Comment:
no, otherwise here you'd keep the error monitor :-)

You always delete both. No need to emit an error packet.